extensions/amp-viqeo-player/0.1/amp-viqeo-player.js

```diff
@@ -0,0 +1,347 @@
+    this.iframe_ = null;
+
+    /** @private {?Promise} */
+    this.playerReadyPromise_ = null;
+
+    /** @private {?Function} */
+    this.playerReadyResolver_ = null;
+
+    /** @private {?number} */
+    this.volume_ = null;
+
+    /** @private {?Function} */
+    this.unlistenMessage_ = null;
+
+    /** @private {?Object} */
+    this.viqeoPlayer_ = null;
+
+    /** @private {boolean} */
+    this.kindIsProd_ = true;
+  }
+
+  /**
+   * @param {boolean=} opt_onLayout
+   * @override
+   */
+  preconnectCallback(opt_onLayout) {
+    this.preconnect.url('https://static.viqeo.tv', opt_onLayout);
+    this.preconnect.url('https://stage.embed.viqeo.tv', opt_onLayout);
+  }
+
+  /**
+   * @param {!Layout} layout
+   * @return {boolean}
+   * @override
+   */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
+  /** @override */
+  createPlaceholderCallback() {
+    return null;
+  }
+
+  /** @override */
+  buildCallback() {
+    this.videoId_ = user().assert(
+        this.element.getAttribute('data-videoid'),
+        'The data-videoid attribute is required for <amp-viqeo-player> %s',
+        this.element);
+
+    this.profileId_ = user().assert(
+        this.element.getAttribute('data-profileid'),
+        'The data-profileid attribute is required for <amp-viqeo-player> %s',
+        this.element);
+
+    this.markTagParams_ = this.element.getAttribute('data-tag-settings');
+
+    this.kindIsProd_ = this.element.getAttribute('data-kind') !== 'stage';
+
+    const deferred = new Deferred();
+    this.playerReadyPromise_ = deferred.promise;
+    this.playerReadyResolver_ = deferred.resolve;
+
+    installVideoManagerForDoc(this.element);
+    Services.videoManagerForDoc(this.element).register(this);
+
+  }
+
+  /** @override */
+  layoutCallback() {
+    let scriptPlayerInit = this.element.getAttribute('data-script-url');
+    scriptPlayerInit =
+        (scriptPlayerInit
+            && tryDecodeUriComponent(scriptPlayerInit)
+        )
+        ||
+        (this.kindIsProd_
+          ? 'https://cdn.viqeo.tv/js/vq_player_init.js?amp=true'
+          : 'https://static.viqeo.tv/js/vq_player_init.js?branch=dev1&amp=true'
+        );
+    // embed preview url
+    let previewUrl = this.element.getAttribute('data-player-url');
+    previewUrl =
+        (previewUrl
+            && previewUrl.length && decodeURI(previewUrl)
+        )
+        || (this.kindIsProd_ ? 'https://cdn.viqeo.tv/embed' : 'https://stage.embed.viqeo.tv');
+
+    // Create preview iframe source path
+    previewUrl = assertAbsoluteHttpOrHttpsUrl(
+        `${previewUrl}/?vid=${this.videoId_}&amp=true`);
+
+    const jsonParams = {
+      scriptPlayerInit: encodeURIComponent(scriptPlayerInit),
+      previewUrl: encodeURIComponent(previewUrl),
+      videoId: this.videoId_,
+      profileId: this.profileId_,
+      markTagParams: this.markTagParams_,
+    };
+
+    const iframe = getIframe(this.win, this.element, 'viqeoplayer', jsonParams);
```

Comment:
I test your player, and `fullscreen` did not work. You would need to pass `{allowFullscreen: true}` as last arg to the `getIframe` call.